vividus-extension-selenium/src/main/java/org/vividus/proxy/Proxy.java

```diff
@@ -144,8 +115,17 @@ public void clearRequestFilters()
         return ClientUtil.createSeleniumProxy(proxyServer, InetAddress.getLoopbackAddress());
     }
 
-    public void setProxyServerFactory(IProxyServerFactory proxyServerFactory)
+    BrowserUpProxy getProxyServer()
     {
-        this.proxyServerFactory = proxyServerFactory;
+        return proxyServer;
+    }
+
+    <R> R executeIfProxyStarted(Function<BrowserUpProxy, R> proxyAction)
```

Comment:
why the executeIfProxyStarted and getProxyServer package private?

Reply:
`getProxyServer` is used by `ThreadedProxy`
`executeIfProxyStarted` was updated to `private`